sklearn/utils/tests/test_testing.py

```diff
@@ -84,10 +90,181 @@ def _raise_ValueError(message):
                   _raise_ValueError, "test")
 
 
+class MockClassifier():
+    def __init__(self, same_predict=True, same_transform=True,
+                 same_decision_function=True, same_predict_proba=True,
+                 same_attributes=True, n_outputs=1):
+        self._same_predict = same_predict
+        self._same_transform = same_transform
+        self._same_decision_function = same_decision_function
+        self._same_predict_proba = same_predict_proba
+        self._n_outputs = n_outputs
+        self._same_attributes = same_attributes
+
+    def fit(self, X=None, y=None):
+        X, y = np.array(X), np.array(y)
+        self._n_features = X.shape[1] if len(X.shape) == 2 else 1
+        self._n_classes = np.unique(y).size
+        if self._same_attributes:
+            random_base_module = np.random.RandomState(0)  # Seeded RNG
+        else:
+            random_base_module = np.random  # Unseeded RNG
+        # A float attribute
+        self.attribute1_ = random_base_module.random_sample()
+        # A list attribute
+        self.attribute2_ = random_base_module.random_sample((10, 10))
+        return self
+
+    def _output(self, X, same=False, integer=True, shape=None):
+        X = np.array(X)
+        n_samples = X.shape[0] if len(X.shape) == 2 else X.size
+        if shape is None:
+            shape = (n_samples, self._n_outputs)
+        if same:
+            random_base_module = np.random.RandomState(0)
+        else:
+            random_base_module = np.random  # Truly random
+        if integer:
+            return random_base_module.randint(0, self._n_classes, size=shape)
+        else:
+            return random_base_module.random_sample(shape)
+
+    def predict(self, X):
+        return self._output(X, self._same_predict, integer=True)
+
+    def transform(self, X):
+        return self._output(X, self._same_transform, integer=False)
+
+    def predict_proba(self, X):
+        return self._output(X, self._same_predict_proba, integer=False)
+
+    def decision_function(self, X):
+        return self._output(X, self._same_decision_function, integer=False)
+
+
+def test_assert_xsame_model_():
```

Comment:
Indeed, still I find the name confusing. Maybe `test_assert_same_not_same_model` is better.